src/Dscp/Core/Types.hs

```diff
@@ -223,3 +276,129 @@ newtype ATG = ATG
 makeLenses ''ATGNode
 makeLenses ''ATGEdge
 makeLenses ''ATG
+
+
+----------------------------------------------------------------------------
+-- Transactions
+----------------------------------------------------------------------------
+
+-- We have several different types of "transactions". Money
+-- transactions, delegation transactions (in the future). Money
+-- transactions are the most popular, so we'll call them just
+-- "transactions".
+
+-- | Money transaction input.
+data TxIn = TxIn
+    { txInAddr  :: Address
+    , txInNonce :: Word64
+    , txInValue :: Coin
+    } deriving (Eq, Ord, Generic, Show)
+
+instance Buildable TxIn where
+    build TxIn{..} = "<" +|| txInAddr ||+ " w nonce " +|| txInNonce ||+ ", " +|| txInValue ||+ ">"
+
+-- | Money transaction output.
+data TxOut = TxOut
+    { txOutAddr  :: Address
+    , txOutValue :: Coin
+    } deriving (Eq, Ord, Generic, Show)
+
+instance Buildable TxOut where
+    build TxOut{..} = "<" +|| txOutAddr ||+ ", " +|| txOutValue ||+ ">"
+
+-- | Transaction. Accounting-style. TODO add other types.
+data Tx = Tx
+    { txIns  :: [TxIn]
+    , txOuts :: [TxOut]
+    } deriving (Eq, Ord, Generic, Show)
+
+instance Buildable Tx where
+    build Tx{..} = "Tx { in: " +|| listF txIns ||+ "; outs:" +|| listF txOuts ||+ " }"
+
+type TxId = Hash Tx
+
+-- | Compute tx id.
+txId :: (Serialise Tx) => Tx -> TxId
```

Comment:
We have `HasId` class (in `Dscp.Util`) for dealing with such stuff uniformly (though it probably got merged in `master` before the last time you rebase onto master)

Reply:
It's not related to `HasId` at all, `TxId` is a transaction hash.

Reply:
It has - I use transaction hash as its database `Id`.

Reply:
That's not a point. You can implement `HasId` instance in the database module if you need it, but `TxId` is a separate type, which is not related to the database keys in general.